main/php_network.h

```diff
@@ -229,7 +232,7 @@ PHPAPI void php_network_freeaddresses(struct sockaddr **sal);
 
 PHPAPI php_socket_t php_network_connect_socket_to_host(const char *host, unsigned short port,
 		int socktype, int asynchronous, struct timeval *timeout, char **error_string,
-		int *error_code, char *bindto, unsigned short bindport 
+		int *error_code, char *bindto, unsigned short bindport, long sockopts
 		TSRMLS_DC);
```

Comment:
This is effectively an ABI break.
I would suggest NOT to introduce new _ex functions in PHP 5 branch , please, target PHP7

If you target PHP7 and rebase your PR against it, I will be pleased to merge it :-)